components/sysutils/xca/Makefile

```diff
@@ -0,0 +1,62 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2024, Friedrich Kink. All rights reserved.
+#
+
+BUILD_STYLE= cmake
+USE_PARALLEL_BUILD= yes
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		xca
+COMPONENT_VERSION=	2.6.0
+COMPONENT_SUMMARY=	XCA - X Certificate and Key Management
+COMPONENT_DESCRIPTION=	This application is intended for creating and managing \
+			X.509 certificates, certificate requests, RSA, DSA and EC \
+			private keys, Smartcards and CRLs.\
+			Everything that is needed for a CA is implemented. \
+			All CAs can sign sub-CAs recursively. These certificate chains \
+			are shown clearly.
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:55a39f22eabadcce4532ebfcf07f796372bdb0d323720db6b6e0deaf267b1709
+COMPONENT_PROJECT_URL=	http://xca.hohnstaedt.de
+COMPONENT_ARCHIVE_URL=	https://github.com/chris2511/$(COMPONENT_NAME)/releases/download/RELEASE.$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_FMRI= 	sysutils/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=	Applications/System Utilities
+COMPONENT_LICENSE=	Christian Hohnstaedt
+COMPONENT_LICENSE_FILE=	COPYRIGHT
+
+include $(WS_MAKE_RULES)/common.mk
+
+# partly tests need a graphical desktop
+TEST_TARGET= $(NO_TESTS)
+
```

Comment:
We have a solution for this:
# Tests are run using an X virtual framebuffer.
COMPONENT_TEST_ENV= DISPLAY=:1

COMPONENT_PRE_TEST_ACTION = \
    $(SHELL) $(COMPONENT_DIR)/start_Xvfb $(COMPONENT_DIR)
COMPONENT_POST_TEST_ACTION = \
    $(SHELL) $(COMPONENT_DIR)/stop_Xvfb $(COMPONENT_DIR)

# Test dependencies
TEST_REQUIRED_PACKAGES += x11/server/xvfb

with the needed scripts (you can find them eg. in library/gtk+4/

Reply:
Great I implemented, but unfortunately the first test expects interactive input, so fails anyway.
There is still a test failing which I don't understand. To compare time outputs it looks for the information about timezones in /usr/share/zoneinfo whereas Solaris/Illumos has this information in /usr/share/lib/zoneinfo. Does anyone know whether this is a configurable path?

Reply:
When creating a link from /usr/share/lib/zoneinfo /usr/share/zoneinfo the test passes.

Reply:
i found the issue:
strings /usr/lib/qt/5.15/lib/amd64/libQt5Core.so.5.15.13|grep zoneinfo
/usr/share/zoneinfo/
/usr/lib/zoneinfo/

Qt5 is the guilty package.

Reply:
Now that qt5 is updated and the zoneinfo patch is included this package can be released as everything is working (incl. all tests).